src/m365/app/commands/permission/permission-add.spec.ts

```diff
@@ -0,0 +1,321 @@
+  const applications: Application[] = [{ 'id': appId, 'requiredResourceAccess': [] }];
+  const applicationPermissions = 'https://graph.microsoft.com/User.ReadWrite.All https://graph.microsoft.com/User.Read.All';
+  const delegatedPermissions = 'https://graph.microsoft.com/offline_access';
+  //#endregion
+
+  let log: string[];
+  let logger: Logger;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    sinon.stub(session, 'getId').callsFake(() => '');
+    sinon.stub(fs, 'existsSync').callsFake(() => true);
+    sinon.stub(fs, 'readFileSync').callsFake(() => JSON.stringify({
+      apps: [
+        {
+          appId: appId,
+          name: 'CLI app1'
+        }
+      ]
+    }));
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.patch,
+      request.post,
+      odata.getAllItems
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName,
+      session.getId,
+      fs.existsSync,
+      fs.readFileSync
+    ]);
```

Comment:
Hi, just popping in here to say that we should refactor this `sinonUtil.restore([...` to `sinon.restore()`. This is a new way to ensure that the entire sandbox is cleared and will from now on be used in every command.